pkg/apis/core/validation/shoot.go

```diff
@@ -2180,22 +2180,25 @@ func validateHAShootControlPlaneConfigurationValue(shoot *core.Shoot) field.Erro
 	return allErrs
 }
 
-func validateShootHAControlPlaneSpecUpdate(newSpec, oldSpec *core.ShootSpec, fldPath *field.Path) field.ErrorList {
+func validateShootHAControlPlaneSpecUpdate(newShoot, oldShoot *core.Shoot, fldPath *field.Path) field.ErrorList {
 	var (
 		allErrs          = field.ErrorList{}
-		shootIsScheduled = newSpec.SeedName != nil
+		shootIsScheduled = newShoot.Spec.SeedName != nil
 
 		oldVal, newVal core.FailureToleranceType
 		oldValExists   bool
 	)
 
-	if oldSpec.ControlPlane != nil && oldSpec.ControlPlane.HighAvailability != nil {
-		oldVal = oldSpec.ControlPlane.HighAvailability.FailureTolerance.Type
+	if oldShoot.Spec.ControlPlane != nil && oldShoot.Spec.ControlPlane.HighAvailability != nil {
+		oldVal = oldShoot.Spec.ControlPlane.HighAvailability.FailureTolerance.Type
 		oldValExists = true
 	}
 
-	if newSpec.ControlPlane != nil && newSpec.ControlPlane.HighAvailability != nil {
-		newVal = newSpec.ControlPlane.HighAvailability.FailureTolerance.Type
+	if newShoot.Spec.ControlPlane != nil && newShoot.Spec.ControlPlane.HighAvailability != nil {
+		newVal = newShoot.Spec.ControlPlane.HighAvailability.FailureTolerance.Type
+		if !oldValExists && isShootInHibernation(newShoot) {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "controlPlane", "highAvailability", "failureTolerance", "type"), "Shoot is currently hibernated and cannot be scaled up to HA. Please make sure your cluster has woken up before scaling it up to HA"))
```

Comment:
spec and controlPlane are already part of the field path passed. Use `fldPath.Child`